src/components/CallNotify.tsx

```diff
@@ -1,69 +1,179 @@
-import { mdiCheck, mdiClose } from '@mdi/js'
 import { observer } from 'mobx-react'
 import React from 'react'
-import { StyleSheet, View } from 'react-native'
+import { Image, StyleSheet, TouchableOpacity, View } from 'react-native'
+import UserAvatar from 'react-native-user-avatar'
 
 import callStore from '../stores/callStore'
+import contactStore from '../stores/contactStore'
 import intl from '../stores/intl'
-import ButtonIcon from './ButtonIcon'
+import CustomGradient from './CustomGradient'
 import { RnText } from './Rn'
-import g from './variables'
 
 const css = StyleSheet.create({
   Notify: {
+    flex: 1,
+  },
+  NotifyContainer: {
+    flex: 1,
+    marginBottom: 100,
+  },
+  Notify_Btn_Side_By_Side: {
+    flex: 1,
     flexDirection: 'row',
-    alignItems: 'center',
-    borderBottomWidth: 1,
-    borderColor: g.borderBg,
-    backgroundColor: g.hoverBg,
+    justifyContent: 'space-around',
+    alignItems: 'flex-end',
   },
   Notify_Info: {
-    flex: 1,
-    paddingLeft: 12,
-    paddingVertical: 5,
+    alignItems: 'center',
+    justifyContent: 'center',
+    marginTop: '30%',
   },
   Notify_Btn_reject: {
-    borderColor: g.colors.danger,
+    width: 80,
+    height: 80,
   },
   Notify_Btn_accept: {
-    borderColor: g.colors.primary,
+    width: 80,
+    height: 80,
+  },
+  MobileNumber: {
+    color: '#2276FF',
+    fontSize: 25,
+    lineHeight: 28,
+  },
+  UserNumber: {
+    color: '#2F3443',
+    fontSize: 25,
+    lineHeight: 28,
+  },
+  FlagLogo: {
+    width: 20,
+    height: 14,
+    marginRight: 10,
+  },
+  ActionBtnText: {
+    fontSize: 16,
+    color: '#2F3443',
+    marginTop: 7,
+  },
+  PoweredBy: {
+    alignItems: 'center',
+    flexDirection: 'row',
+    justifyContent: 'center',
+    marginTop: 40,
+  },
+  PoweredByText: {
+    color: '#BCC4CC',
+    fontSize: 11,
+  },
+  qooqieLogo: {
+    width: 56.3,
+    height: 14,
+    marginLeft: 10,
+  },
+  actionBtnContainer: {
+    alignItems: 'center',
+  },
+  CallerName: {
+    color: '#2F3443',
+    fontSize: 26,
+    lineHeight: 28,
+    margin: 7,
   },
 })
 
 @observer
 class CallNotify extends React.Component {
+  state = {
+    callerName: '',
+  }
   render() {
     const c = callStore.incomingCall
     if (!c || callStore.recentPn?.action) {
       return null
     }
+
+    if (c.partyName && c.partyName != c.partyNumber) {
+      this.setState({ callerName: c.partyName })
+    } else {
+      contactStore.getPartyName(c.partyNumber, (value: String) =>
+        this.setState({ callerName: value }),
+      )
+    }
+    let callerNumber = c.partyNumber
```

Comment:
callerNumber can be declared as const as we do not change the value of it in the component.